crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -371,10 +381,15 @@ where
 
     pub fn new_link(
         url: S,
+        attributes: Option<Vec<(S, S)>>,
         children: Vec<DomNode<S>>,
         mention_type: Option<S>,
     ) -> Self {
-        let mut attrs = vec![("href".into(), url.clone())];
+        // FIXME: mention_type could be removed, and the hosting application could just provide attributes
+        // this would allow the Rust code to stay as generic as possible, since it should only care abouy
+        // `contenteditable="false"` to implement custom behaviours for immutable links.
+        let mut attrs =
+            attributes.unwrap_or_else(|| vec![("href".into(), url.clone())]);
```

Comment:
This is a really good idea - this could be used to add classes and styles to links in the markup which I think will be really helpful with implementing avatars on web

Reply:
One important thing to note, is that these attributes **will** end up in the final HTML in order for message edition to work and re-create pills, so we still need to be careful about what ends up in the HTML or not.